rest-api/tools/import_organizations.py

```diff
@@ -201,14 +216,28 @@ def _cleanup_old_entities(self, session, row_list, dry_run):
     if orgs_to_remove:
       org_id_list = []
       for org in orgs_to_remove:
-        log_prefix = '(dry run) ' if dry_run else ''
-        logging.info(log_prefix + 'Deleting old Organization no longer in Google sheet: %s', org)
         old_org = self.org_dao.get_by_external_id(org)
-        org_id_list.append(old_org.organizationId)
-        self.deletion_count += 1
+        if old_org and old_org.isObsolete != ObsoleteStatus.OBSOLETE:
+          org_id_list.append(old_org.organizationId)
+          self.deletion_count += 1
+        elif old_org.isObsolete == ObsoleteStatus.OBSOLETE:
+          logging.info('Not attempting to delete org [%s] with existing obsolete status',
+                       old_org.displayName)
 
       if org_id_list and not dry_run:
-        self.delete_sql_statement(session, org_id_list)
+        logging.info(log_prefix + 'Marking old Organization as obsolete referenced in other '
+                                  'table: %s', old_org)
+        str_list = ','.join([str(i) for i in org_id_list])
+        sql = """ UPDATE organization
+            SET is_obsolete = 1
+            WHERE organization_id in ({org_id_list})""".format(org_id_list=str_list)
+        session.execute(sql)
+
+        self.org_dao._invalidate_cache()
+        # Try to delete old orgs.
+        logging.info(log_prefix + 'Deleting old Organization no longer in Google sheet: %s', org)
+        self.delete_sql_statement(session, str_list)
```

Comment:
Same as above, might do this before invalidating cache.

Reply:
done